modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/SqlQueryProcessor.java

```diff
@@ -620,4 +629,10 @@ public CompletableFuture<Boolean> notify(@NotNull IndexEventParameters parameter
                     .thenApply(v -> false);
         }
     }
+
+    /** Returns {@code true} if this is data modification operation. */
+    private static boolean dataModificationOp(SqlNode sqlNode) {
+        return sqlNode instanceof SqlInsert || sqlNode instanceof SqlMerge || sqlNode instanceof SqlDelete
```

Comment:
probably it would be better to replace this with `SqlKind.DML.contains(sqlNode.getKind())`